### PR TITLE
Simplify symbolize_pprof

### PR DESCRIPTION
**Pull request by nsavoire**
# What does this PR do?

Simplify a bit `symbolize_pprof` function.

### CODE REVIEW

File: src/symbolizer.cc [hunk 1]
```diff
@@ -45,49 +59,38 @@ DDRes Symbolizer::symbolize_pprof(std::span<ElfAddress_t> elf_addrs,
...
     // Symbolize the addresses
-    blaze_res = blaze_symbolize_elf_virt_offsets(
-        symbolizer, &src_elf, elf_addrs.data(), elf_addrs.size());
+    const auto *blaze_res = blaze_symbolize_elf_virt_offsets(
+        symbolizer_wrapper.symbolizer.get(), &src_elf, elf_addrs.data(),
+        elf_addrs.size());
+    if (!blaze_res && symbolizer_wrapper.use_debug) {
+      // Symbolization failed, retry without using debug symbols
+      // blazesym curently does not support compressed debug sections:
+      // cf. https://github.com/libbpf/blazesym/issues/581
```
Comment (r1viollet) [inline src/symbolizer.cc:85]: ah yeah, nice adding the reference

File: src/symbolizer.cc [hunk 2]
```diff
@@ -45,49 +59,38 @@ DDRes Symbolizer::symbolize_pprof(std::span<ElfAddress_t> elf_addrs,
...
         .reserved = {},
     };
 
     // Symbolize the addresses
-    blaze_res = blaze_symbolize_elf_virt_offsets(
-        symbolizer, &src_elf, elf_addrs.data(), elf_addrs.size());
+    const auto *blaze_res = blaze_symbolize_elf_virt_offsets(
+        symbolizer_wrapper.symbolizer.get(), &src_elf, elf_addrs.data(),
+        elf_addrs.size());
+    if (!blaze_res && symbolizer_wrapper.use_debug) {
```
Comment (r1viollet) [inline src/symbolizer.cc:82]: Do you want to change to a retry at every symbolization attempt ?

**Review by r1viollet — APPROVED** [review]
LGTM

File: src/symbolizer.cc (hunk 2 above)
Comment (nsavoire) [reply src/symbolizer.cc:82]: Hum, it should only retry on the first symbolization attempt:
* on first attempt `symbolizer_wrapper.use_debug` is `true`, so symbolizer is called with debug symbols
* if it fails, `symbolizer_wrapper.use_debug` is set to false and symbolizer is called again without debug symbols
* on next attempt, `symbolizer_wrapper.use_debug` is false, so symbolizer is called without debug symbols and no other attempt is made is it fails

Comment (r1viollet) [reply src/symbolizer.cc:82]: That makes sense :+1:

**Pull request merged**